modules/utils.js

```diff
@@ -112,6 +112,19 @@ const collectParams = ({ tableHeader, tableBody }) => {
   }, {});
 };
 
+const findAstNodesData = (testSteps) => {
+  const isRule = testSteps.some((entity) => 'rule' in entity);
+
+  if (!isRule) {
+    return testSteps.reduce((acc, steps) => {
+      const keywords = Object.keys(steps);
+      return acc.concat(...keywords.map((keyword) => steps[keyword].steps));
+    }, []);
+  }
+
+  return findAstNodesData(testSteps.reduce((acc, { rule }) => acc.concat(rule.children), []));
+};
```

Comment:
If I correctly understands this function, its requires the flat structure of `testSteps` (actually the feature's `children`) to collect their `keywords` correctly.
So if we never have more than one nesting for grouping entities like `rule`, may be better simplify the logic here with two main steps:
1) Make the flat structure for feature's `children` with their own `children`.
2) Make the flat structure for `steps` from all entities.

F.e.
```javascript
const collectAstNodesData = (children) => {
    const flattenChildren = children.reduce((acc, child) => acc.concat('rule' in entity ? child.rule.children : child), []);

    return flattenChildren.reduce((acc, child) => {
      const childValues = Object.values(child); // array like [Scenario obj, Background obj]
      return acc.concat(childValues.map((childValue) => childValue.steps).flat());
    }, []);
};
```